implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/v2/DynatraceExporterV2Test.java

```diff
@@ -595,6 +602,247 @@ public String get(String key) {
+            public String uri() {
+                return "http://localhost";
+            }
+
+            @Override
+            @SuppressWarnings("NullableProblems")
+            public String apiToken() {
+                return "apiToken";
+            }
+
+            @Override
+            @SuppressWarnings("NullableProblems")
+            public DynatraceApiVersion apiVersion() {
+                return DynatraceApiVersion.V2;
+            }
+
+            @Override
+            public int batchSize() {
+                return 3;
+            }
+        };
+
+        DynatraceExporterV2 exporter = new DynatraceExporterV2(config, clock, httpClient);
+        DynatraceMeterRegistry meterRegistry = DynatraceMeterRegistry.builder(config)
+            .httpClient(httpClient)
+            .clock(clock)
+            .build();
+
+        Counter counter = Counter.builder("my.count")
+            .description("count description")
+            .baseUnit("Bytes")
+            .register(meterRegistry);
+        counter.increment(5.234);
+        Gauge.builder("my.gauge", () -> 1.23).description("my.description").baseUnit("Liters").register(meterRegistry);
+        clock.add(config.step());
+        exporter.export(meterRegistry.getMeters());
+
+        ArgumentCaptor<String> firstReqCap = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> secondReqCap = ArgumentCaptor.forClass(String.class);
+        verify(firstReq).withPlainText(firstReqCap.capture());
+        verify(secondReq).withPlainText(secondReqCap.capture());
+
+        List<String> firstReqLines = Arrays.asList(firstReqCap.getValue().split("\n"));
+        List<String> secondReqLines = Arrays.asList(secondReqCap.getValue().split("\n"));
+
+        // the first request will contain the metric lines
+        assertThat(firstReqLines).hasSize(3)
+            .containsExactly("my.count,dt.metrics.source=micrometer count,delta=5.234 " + clock.wallTime(),
+                    "my.gauge,dt.metrics.source=micrometer gauge,1.23 " + clock.wallTime(),
+                    "#my.count count dt.meta.description=count\\ description,dt.meta.unit=Bytes");
+
+        // the second request will the leftover metadata line
+        assertThat(secondReqLines).hasSize(1)
+            .containsExactly("#my.gauge gauge dt.meta.description=my.description,dt.meta.unit=Liters");
+    }
+
+    @Test
+    void metadataIsSerializedOnceWhenSetTwice() {
+        HttpSender.Request.Builder builder = spy(HttpSender.Request.build(config.uri(), httpClient));
+        when(httpClient.post(anyString())).thenReturn(builder);
+
+        // both counters have the same unit and description, but other tags differ
+        Counter counter1 = Counter.builder("my.count")
+            .description("count description")
+            .baseUnit("Bytes")
+            .tag("counter-number", "counter1")
+            .register(meterRegistry);
+        Counter counter2 = Counter.builder("my.count")
```

Comment:
This can't happen in reality, right? Because the meter registry prevents it?

Reply:
It is possible to have two meters with the same name as long as they have the same type. In fact this is how you set up multiple metric streams for one metric key in Micrometer. Only if the types differ but the names are the same (e.g., having a gauge and a counter, both with the name `my.meter`), a conflict will occur.